src/beanmachine/ppl/inference/monte_carlo_samples.py

```diff
@@ -77,7 +95,21 @@ def get_chain(self, chain: int = 0) -> "MonteCarloSamples":
             raise IndexError("Please specify a valid chain")
 
         samples = {rv: self.get_variable(rv, True)[[chain]] for rv in self}
-        new_mcs = MonteCarloSamples(samples, self.num_adaptive_samples)
+
+        if self.log_likelihoods is None:
+            logll = None
+        else:
+            logll = {
+                rv: self.get_log_likelihoods(rv, True)[[chain]]
+                for rv in self.log_likelihoods
+            }
+
+        new_mcs = MonteCarloSamples(
+            chain_results=samples,
+            num_adaptive_samples=self.num_adaptive_samples,
+            logll_results=logll,
```

Comment:
Our internal linter (Pyre) is complaining about the following typing issue:
> Incompatible parameter type [6]: Expected `Optional[List[Dict[RVIdentifier, torch.Tensor]]]` for 3rd parameter `logll_results` to call `MonteCarloSamples.__init__` but got `Optional[Dict[typing.Any, typing.Any]]`.

Speaking of which, do you mind adding a call to `get_chain` to the unit test as well? Thanks :D

Reply:
That should fix the linter error.